### PR TITLE
update the translation of `Object.valueOf()`

### PR DESCRIPTION
**Pull request by robert-jx**
### Description
update the translation of `Object.valueOf()`

### Motivation
update the translation of `Object.valueOf()`

### Additional details
none

### CODE REVIEW

File: files/zh-cn/web/javascript/reference/global_objects/object/valueof/index.md [hunk 1]
```diff
@@ -21,23 +21,23 @@ valueOf()
 
 ### 返回值
 
-`this` 值，将其转换为一个对象。
+转换成对象的 `this` 值。
 
-> **备注：** 为了使 `valueOf` 在类型转换期间有用，它必须返回一个原始值。因为所有原始类型都有它们自己的 `valueOf()` 方法，调用 `aPrimitiveValue.valueOf()` 通常不会调用 `Object.prototype.valueOf()`。
+> **备注：** 为了使 `valueOf` 在类型转换过程中有用，它必须返回基本类型。因为所有的基本类型都有自己的 `valueOf()` 方法，所以调用 `aPrimitiveValue.valueOf()` 通常不会调用 `Object.prototype.valueOf()`。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/object/valueof/index.md:26]:
```suggestion
> **备注：** 为了使 `valueOf` 在类型转换过程中有用，它必须返回基本类型。因为所有的基本类型都有自己的 `valueOf()` 方法，所以调用 `aPrimitiveValue.valueOf()` 通常不会调用 `Object.prototype.valueOf()`。
```
```suggestion
> **备注：** 为了使 `valueOf` 在类型转换过程中有用，它必须返回一个基本类型值。因为所有的基本类型都有自己的 `valueOf()` 方法，所以调用 `aPrimitiveValue.valueOf()` 通常不会调用 `Object.prototype.valueOf()`。
```

File: files/zh-cn/web/javascript/reference/global_objects/object/valueof/index.md [hunk 2]
```diff
@@ -21,23 +21,23 @@ valueOf()
...
 
 ## 描述
 
-JavaScript 调用 `valueOf` 方法[将对象转换为一个原始值](/zh-CN/docs/Web/JavaScript/Data_structures#强制类型转换)。你很少需要自己去调用 `valueOf` 方法；当遇到需要原始值的对象时，JavaScript 会自己调用它。
+JavaScript 调用 `valueOf` 方法来[将对象转换成基本类型值](/zh-CN/docs/Web/JavaScript/Data_structures#类型强制转换)。你很少需要自己调用 `valueOf` 方法；当遇到需要基本类型值的对象时，JavaScript 会自动的调用该方法。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/object/valueof/index.md:30]:
```suggestion
JavaScript 调用 `valueOf` 方法来[将对象转换成基本类型值](/zh-CN/docs/Web/JavaScript/Data_structures#强制类型转换)。你很少需要自己调用 `valueOf` 方法；当遇到需要基本类型值的对象时，JavaScript 会自动的调用该方法。
```

File: files/zh-cn/web/javascript/reference/global_objects/object/valueof/index.md [hunk 3]
```diff
@@ -47,11 +47,11 @@ console.log(Object.prototype.valueOf.call("primitive"));
 // [String: 'primitive'] (a wrapper object)
 ```
 
-### 为自定义对象重写 valueOf
+### 覆盖自定义对象的 valueOf
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/object/valueof/index.md:50]: 我倾向于将 override 翻译为：对于属性是覆盖；对于方法是重写
```suggestion
### 重写自定义对象的 valueOf
```

File: files/zh-cn/web/javascript/reference/global_objects/object/valueof/index.md [hunk 4]
```diff
@@ -65,31 +65,31 @@ class Box {
...
 
-对象的 `valueOf` 方法通常由 JavaScript 调用，但是你可以自己调用它，如下所示：
+对象的 `valueOf` 方法通常由 JavaScript 调用，但你可以按照如下方式自己调用它：
 
 ```js
 box.valueOf();
 ```
 
-### 在对象上使用一元加运算符
+### 对对象使用一元加号
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/object/valueof/index.md:82]:
```suggestion
### 对对象使用一元加运算符
```

**Review by JasonLamv-t — COMMENTED** [review]
Some suggestions.

File: files/zh-cn/web/javascript/reference/global_objects/object/valueof/index.md [hunk 5]
```diff
@@ -21,23 +21,23 @@ valueOf()
...
+> **备注：** 为了使 `valueOf` 在类型转换过程中有用，它必须返回一个基本类型值。因为所有的基本类型都有自己的 `valueOf()` 方法，所以调用 `aPrimitiveValue.valueOf()` 通常不会调用 `Object.prototype.valueOf()`。
 
 ## 描述
 
-JavaScript 调用 `valueOf` 方法[将对象转换为一个原始值](/zh-CN/docs/Web/JavaScript/Data_structures#强制类型转换)。你很少需要自己去调用 `valueOf` 方法；当遇到需要原始值的对象时，JavaScript 会自己调用它。
+JavaScript 调用 `valueOf` 方法来[将对象转换成基本类型值](/zh-CN/docs/Web/JavaScript/Data_structures#强制类型转换)。你很少需要自己调用 `valueOf` 方法；当遇到需要基本类型值的对象时，JavaScript 会自动的调用该方法。
 
-该方法由[数字转换](/zh-CN/docs/Web/JavaScript/Data_structures#强制数字类型转换)和[原始值转换](/zh-CN/docs/Web/JavaScript/Data_structures#强制原始值转换)优先调用，但是[字符串转换](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/String#字符串强制转换)会优先调用 `toString()`，并且 `toString()` 非常可能返回一个字符串类型（即使原始实现了 {{jsxref("Object.prototype.toString()")}}），所以 `valueOf()` 在这种情况下通常不会被调用。
+该方法由[强制数字类型转换](/zh-CN/docs/Web/JavaScript/Data_structures#强制数字类型转换)和[强制基本类型转换](/zh-CN/docs/Web/JavaScript/Data_structures#强制原始值转换)优先调用，但是[强制字符串转换](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/String#强制字符串转换)会优先调用 `toString()`，并且 `toString()` 很可能返回字符串值（甚至对于 {{jsxref("Object.prototype.toString()")}} 基本实现也是如此），因此在这种情况下不会调用 `valueOf()`。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/object/valueof/index.md:32]:
```suggestion
[强制数字类型转换](/zh-CN/docs/Web/JavaScript/Data_structures#强制数字类型转换)和[强制基本类型转换](/zh-CN/docs/Web/JavaScript/Data_structures#强制原始值转换)优先会调用该方法，而[强制字符串转换](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/String#强制字符串转换)会优先调用 `toString()`，并且 `toString()` 很可能返回字符串值（甚至对于 {{jsxref("Object.prototype.toString()")}} 基本实现也是如此），因此在这种情况下不会调用 `valueOf()`。
```

**Review by yin1999 — COMMENTED** [review]
almost good, just some suggestions ;)

**Review by yin1999 — APPROVED** [review]
Thanks

**Pull request merged**